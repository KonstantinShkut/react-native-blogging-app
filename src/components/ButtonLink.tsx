import React from 'react';
import { Button } from 'react-native-elements';
import { colorPrimary } from '../styles';
import { StyleSheet } from 'react-native';

interface Props {
  title: string;
  color?: string;
  onPress: () => any;
}

export const ButtonLink = ({ title, color, onPress }: Props) => {
  return (
    <Button
      title={title}
      buttonStyle={styles.buttonStyle}
      containerStyle={styles.containerStyle}
      titleStyle={{
        ...styles.titleStyle,
        color: color || colorPrimary,
      }}
      onPress={onPress}
    />
  );
};

const styles = StyleSheet.create({
  buttonStyle: {
    backgroundColor: 'transparent',
    padding: 0,
  },
  containerStyle: {
    padding: 10,
    alignSelf: 'stretch',
  },
  titleStyle: {
    fontSize: 18,
  },
});
