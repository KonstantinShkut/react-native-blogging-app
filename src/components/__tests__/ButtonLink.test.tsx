import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonLink } from '../ButtonLink';

describe('ButtonLink', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <ButtonLink title="title" onPress={() => null} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const component = renderer.create(
      <ButtonLink title="title" color="white" onPress={() => null} />,
    );
    expect(component).toMatchSnapshot();
  });
});
