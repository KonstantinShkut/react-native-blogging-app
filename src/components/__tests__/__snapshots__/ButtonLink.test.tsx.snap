// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "borderRadius": 3,
      "overflow": "hidden",
      "padding": 10,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2196F3",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonLink renders correctly with custom color 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "borderRadius": 3,
      "overflow": "hidden",
      "padding": 10,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;
